### ISSUE TITLE
Custom fields in custom post type not appearing

### ISSUE THREAD

**Issue by iannacone** [WordPress/gutenberg#16333]
hi,
i was looking for a similar issue, but i found nothing. if there was already i apologize in advance.

**Describe the bug**
after creating a custom post type, i am not able to see the custom fields.

**To reproduce**
1. fresh install
2.
```
function register_custom_post_types() {
	register_post_type('test', [
		'labels' => [
			'name' => __('test'),
		],
		'public' => true,
		'supports' => [
			'title',
			'editor',
		],
		'show_in_rest' => true,
	]);
}
add_action('init', 'register_custom_post_types');
```
3. create a new cpt post and try to activate the custom fields in the options modal
4. (forced) refresh
5. nothing appears and the custom fields checkbox still unchecked

**Expected behavior**
the custom fields appear a the bottom of the editing page of the custom post type.

**Screenshots**
![image](https://user-images.githubusercontent.com/17182025/60287983-85385280-9913-11e9-9fe7-c3b7be63a4dc.png)

**Desktop:**
 - OS: windows
 - Browser firefox
 - Version 67.0.4 and 68.0b13

**Additional context**
wordpress 5.2.2

**Comment by Soean** [comment]
You need to add `custom-fields` to the supports array, see docs https://developer.wordpress.org/reference/functions/register_post_type

**Comment by Soean** [comment]
We should only show this option, if the post type supports this feature.

**Comment by iannacone** [comment]
sorry for reporting a non bug. i did not notice about that in the documentation.
thank you for the answer.
is it anyway considered as bug because the custom fields checkbox should not appear at all, or i can close this issue?

**Comment by Soean** [comment]
Yes, it's a bug. The option should be not visible when the post type doesn't support this feature. I opened a PR #16338 with a solution. When it's merged we close this issue..

**Comment by paaljoachim** [comment]
It looks like this issue just stopped up as the PR has stalled.
It would be great with a status update and how we can move the issue and associated PR onward.
Thanks!

**Comment by skorasaurus** [comment]
I believe this has been fixed by #33931 ; the fix should appear in Gutenberg 11.5.

I'm not longer able to reproduce this when using a newest build of Gutenberg that includes the changes from 33931; the custom fields in the preferences does not appear :)
